src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1258,29 +1264,65 @@ CATCH_RETURN();
-                                    collection.Get(),
-                                    familyName.data(),
-                                    weight,
-                                    style,
-                                    stretch,
-                                    &mappedLength,
-                                    &mappedFont,
-                                    &scale);
-
-            RETURN_IF_FAILED(_SetMappedFont(textPosition, mappedLength, mappedFont.Get(), scale));
+            const auto axesCount = format3->GetFontAxisValueCount();
+            std::vector<DWRITE_FONT_AXIS_VALUE> axesVector;
+            axesVector.resize(axesCount);
+            const auto res = format3->GetFontAxisValues(axesVector.data(), axesCount);
 
-            textPosition += mappedLength;
-            textLength -= mappedLength;
+            // Walk through and analyze the entire string
+            while (textLength > 0)
+            {
+                UINT32 mappedLength = 0;
+                ::Microsoft::WRL::ComPtr<IDWriteFontFace5> mappedFont;
+                FLOAT scale = 0.0f;
+
+                fallback1->MapCharacters(source,
+                                         textPosition,
+                                         textLength,
+                                         collection.Get(),
+                                         familyName.data(),
+                                         axesVector.data(),
+                                         axesCount,
+                                         &mappedLength,
+                                         &scale,
+                                         &mappedFont);
+
+                RETURN_IF_FAILED(_SetMappedFontFace(textPosition, mappedLength, mappedFont, scale));
+
+                textPosition += mappedLength;
+                textLength -= mappedLength;
+            }
+        }
+        else
+        {
+            // Walk through and analyze the entire string
+            while (textLength > 0)
+            {
+                UINT32 mappedLength = 0;
+                ::Microsoft::WRL::ComPtr<IDWriteFont> mappedFont;
+                FLOAT scale = 0.0f;
+
+                fallback->MapCharacters(source,
+                                        textPosition,
+                                        textLength,
+                                        collection.Get(),
+                                        familyName.data(),
+                                        weight,
+                                        style,
+                                        stretch,
+                                        &mappedLength,
+                                        &mappedFont,
+                                        &scale);
+
+                RETURN_IF_FAILED(_SetMappedFont(textPosition, mappedLength, mappedFont.Get(), scale));
+
+                textPosition += mappedLength;
+                textLength -= mappedLength;
+            }

```

Comment:
This section definitely needs some cleaning up with all the repeated code, wanted to make sure this is the correct way of doing it at all first